### PR TITLE
Backporting improvements in L and TM-framework

### PR DESCRIPTION
**Pull request by fakusb**
This ports improvements for L and TM framework to merge back into undecidability repository.

This PR is work in progress and gets tidied up more. I want to have a draft here so CI runs.

### CODE REVIEW

File: theories/_CoqProject [hunk 1]
```diff
@@ -331,20 +331,19 @@ TM/Univ/StepTM.v
...
-# TM/PrettyBounds/UnivBounds.v
-# TM/PrettyBounds/M2MBounds.v
+TM/PrettyBounds/BaseCode.v
+TM/PrettyBounds/UnivBounds.v
+TM/PrettyBounds/M2MBounds.v
 
-# TM/PrettyBounds/SpaceBounds.v
-# TM/PrettyBounds/BaseCodeSpace.v
-# TM/PrettyBounds/UnivSpaceBounds.v
+TM/PrettyBounds/SpaceBounds.v
```
Comment (yforster) [inline theories/_CoqProject:338]: Are these files really needed for compilation? On the long run we shouldn't have such proofs in the undecidability library. Even on short term it might make sense to already now move them to the complexity library.

Comment (fakusb) [reply theories/_CoqProject:338]: Yes, we should move them out.

**Comment by fakusb** [comment]
So, I removed all files concerning "nice bounds" on TMs.

**Comment by yforster** [comment]
Thanks! Once CI says yes I'll merge
